srcs/main/prompt.c

```diff
@@ -6,27 +6,55 @@
 /*   By: jbarbay <jbarbay@student.42singapore.sg    +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/01/12 18:32:31 by jbarbay           #+#    #+#             */
-/*   Updated: 2024/01/15 22:38:57 by jbarbay          ###   ########.fr       */
+/*   Updated: 2024/01/21 13:32:58 by jbarbay          ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 // Read line
 // Exit when type exit
-// History
+// Add to history (except if the line is empty)
+// Create tokens based on the input
+// If tokens is NULL, it means there was an error (with message): new prompt
 
 #include "../../includes/minishell.h"
 
+int	is_exit(char *line)
+{
+	char	*trimmed;
+
+	trimmed = ft_strtrim(line, WHITESPACE);
+	if (ft_strncmp(trimmed, "exit", 5) == 0)
+	{
+		free(trimmed);
+		return (1);
+	}
+	free(trimmed);
+	return (0);
+}
+
 int main(void)
 {
 	char	*line;
+	t_token	*tokens;
 
 	while (1)
```

Comment:
while (data.exit_flag)

data->exit_flag = 0 in my ft_exit

Reply:
The exit flag is for what?

Reply:
We need a flag to do a while loop until we get an "exit" command as all of the Bash builtins return an exit status of zero if they succeed and a non-zero status on failure. So I don't want to do exit(1) inside my function exit.c

And also I don't think that the way you handle just "exit" without params is a good practice for same reason but its ok for me (I would prefer to call function anyway)

Reply:
Yeah for the exit I used, I created that to be able to test the shell easily. 
But we still need the exit built in. Once the exit built-in is ready we could delete and replace it with your builtin.
Because mine doesn't handle a case where you have pipes / other commands and exit later, or exit status

Reply:
Also, we have to make sure to free everything before we exit. Right now everything is being freed in main, so you need to have access to everything to free